### DISCUSSION TITLE
How to add Modal Parameters using ShowAsync<T>(string title, Dictionary<string, object> parameters = null)

### DISCUSSION THREAD

**Discussion by YouriOkken** [vikramlearning/blazorbootstrap#971] [Q&A]
Hi, I am new to Blazor Bootstrap. I have a very simple question.

How do I pass Modal Parameters such as HeaderCssClass, IsVerticallyCentered, etc. using the ShowAsync(string title, Dictionary<string, object> parameters = null)?

I cant find any info about this on the docs

Kind regards

**Comment by gvreddy04** [comment]
@YouriOkken Thank you for trying BlazorBootstrap. We will add more examples.

## Example:
```cshtml
<Modal @ref="modal" title="Modal title" IsVerticallyCentered="true" HeaderCssClass="custom-modal-header">
    <BodyTemplate>
        . . . 
    </BodyTemplate>
    <FooterTemplate>
        . . . 
    </FooterTemplate>
</Modal>
```

Link: https://demos.blazorbootstrap.com/modals#vertically-centered

**Reply by YouriOkken** [reply]
Hello Vikram Reddy, thank you for your reply. I knew this approach, but I want to make the modals dynamic. And as I said, the docs provided me with the info I shared. Is there any way I can pass the Modal Parameters while making the modal dynamic? As shown here:
https://demos.blazorbootstrap.com/modals#pass-event-callbacks-to-a-dynamic-component

**Comment by gvreddy04** [comment]
How to add Modal Parameters using ShowAsync<T>(string title, Dictionary<string, object> parameters = null) #970
